core/blocks.go

```diff
@@ -18,8 +18,10 @@ package core
 
 import "github.com/dominant-strategies/go-quai/common"
 
-// BadHashes represent a set of manually tracked bad hashes (usually hard forks)
-var BadHashes = map[common.Hash]bool{
-	common.HexToHash("05bef30ef572270f654746da22639a7a0c97dd97a7050b9e252391996aaeb689"): true,
-	common.HexToHash("7d05d08cbc596a2e5e4f13b80a743e53e09221b5323c3a61946b20873e58583f"): true,
+type HeirarchyBadHashes struct {
```

Comment:
Your commit message says you always point back to a prior prime. Which means the same hash will exist in all orders of a given slice. Doesn't `BadHashes` instead need to span slices not order?